src/parameter_input.cpp

```diff
@@ -876,6 +901,35 @@ void ParameterInput::ForwardNextTime(Real mesh_time) {
   }
 }
 
+void ParameterInput::CheckRequired(std::string block, std::string name) {
+  bool missing = true;
+  if (DoesParameterExist(block, name)) {
+    missing = (GetComment(block, name) == "# Default value added at run time");
+  }
+  if (missing) {
+    std::stringstream ss;
+    ss << std::endl
+       << "### ERROR in CheckRequired:" << std::endl
+       << "Parameter file missing required field <" << block << ">/" << name << std::endl
+       << std::endl;
+    throw std::runtime_error(ss.str());
+  }
+}
+
+void ParameterInput::CheckDesired(std::string block, std::string name) {
+  bool missing = true;
+  if (DoesParameterExist(block, name)) {
+    missing = (GetComment(block, name) == "# Default value added at run time");
+  }
+  if (missing) {
+    std::cout << std::endl
+              << "### WARNING in CheckDesired:" << std::endl
+              << "Parameter file missing desired field <" << block << ">/" << name
+              << std::endl
+              << std::endl;
```

Comment:
If we are not requiring certain attributes in the input file and are using default values, I think it would make sense to also print the default values to help the user know what is going on behind the scenes.

Reply:
This is a good idea. Maybe should do the same for the `GetIfExists` family of functions.

Reply:
Second

Reply:
Second

Reply:
good idea!  done.